application/lib/view/screens/sign_up_screen.dart

```diff
@@ -1 +1,146 @@
+import 'package:application/providers/sign_up_view_model_provider.dart';
+import 'package:application/utils/image_src.dart';
+import 'package:application/utils/styles/styles.dart';
+import 'package:application/utils/translation_en.dart';
+import 'package:application/view/widgets/auth_image_widget.dart';
+import 'package:application/view/widgets/custom_circular_progress_indicator.dart';
+import 'package:application/view/widgets/custom_elevated_button.dart';
+import 'package:application/view/widgets/auth_input_decoration.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 
+class SignUpScreen extends ConsumerStatefulWidget {
+  const SignUpScreen({super.key});
+
+  @override
+  ConsumerState<SignUpScreen> createState() {
+    return _SignUpScreenState();
+  }
+}
+
+class _SignUpScreenState extends ConsumerState<SignUpScreen> {
+  final _formKey = GlobalKey<FormState>();
+
+  var _enteredFirstName = '';
+  var _enteredEmail = '';
+  var _enteredPassword = '';
+
+  @override
+  Widget build(BuildContext context) {
+    final viewModel = ref.watch(signUpViewModelProvider);
```

Comment:
A viewmodel használatához használjuk a Provider packaget, ahogy a példában is volt amit küldtem. Így nem kell külön implemetálni egyéni providereket, mint amit most tettél. 
így kell kinéznie a véglegesnek: `final signUpScreenViewModel = Provider.of<signUpScreenViewModel>(context);`

https://nureddineraslan.medium.com/understanding-state-management-with-provider-in-flut-e74e0b9e49d9